tapgarden/custodian.go

```diff
@@ -590,6 +603,27 @@ func (c *Custodian) setReceiveCompleted(event *address.Event,
 	)
 }
 
+// updateReceivedProofs is called by the re-org watcher when it detects a re-org
+// and has updated the received proofs.
+func (c *Custodian) updateReceivedProofs(proofs []*proof.Proof) error {
+	// Let's not be interrupted by a shutdown.
+	ctxt, cancel := c.CtxBlocking()
+	defer cancel()
+
+	headerVerifier := GenHeaderVerifier(ctxt, c.cfg.ChainBridge)
+	for idx := range proofs {
+		err := proof.ReplaceLastProofInBlob(
```

Comment:
I think the case also applies here: recv, transfer, then re-org happens.

Reply:
As in we transfer while the recv is 0-conf, or something like a 1-block reorg that unconfirms the original recv anchor? Or the worse case of a 2+-block reorg unconfirming both anchors.

Wondering if we'd also need to track this for the input UTXOs for minting, since those could also be 1-conf and we'd have the same situation.